dev/com.ibm.ws.ejbcontainer/src/com/ibm/ws/ejbcontainer/osgi/internal/EJBRuntimeImpl.java

```diff
@@ -260,16 +260,21 @@ private static RuntimeException rethrow(Throwable t) {
 
     private final CheckpointPhase checkpointPhase;
 
-    @Activate
-    public EJBRuntimeImpl(@Reference(cardinality = ReferenceCardinality.OPTIONAL) CheckpointPhase checkpointPhase) {
-        if (checkpointPhase != null || !ProductInfo.getBetaEdition()) {
-            this.checkpointPhase = checkpointPhase;
+    public EJBRuntimeImpl() {
+        CheckpointPhase phase = CheckpointPhase.getPhase();
+        if (!ProductInfo.getBetaEdition()) {
+            this.checkpointPhase = phase;
         } else {
-            this.checkpointPhase = CheckpointPhase.getPhase(System.getProperty("io.openliberty.ejb.checkpoint.phase", ""));
+            CheckpointPhase testPhase = CheckpointPhase.getPhase(System.getProperty("io.openliberty.ejb.checkpoint.phase", ""));
+            if (phase == CheckpointPhase.INACTIVE && testPhase != null) {
+                this.checkpointPhase = testPhase;
+            } else {
+                this.checkpointPhase = phase;
+            }
         }
 
         // For any Checkpoint phase, pause all non-persistent timers until checkpoint restored
-        if (checkpointPhase != null && !checkpointPhase.restored()) {
+        if (!checkpointPhase.restored()) {
```

Comment:
The check for `phase != CheckpointPhase.INACTIVE` can be thought of as a replacement for `phase != null`.  

So I think the same code could have been:

```java
        CheckpointPhase checkpointPhase = CheckpointPhase.getPhase();
        if (checkpointPhase != CheckpointPhase.INACTIVE || !ProductInfo.getBetaEdition()) {
            this.checkpointPhase = checkpointPhase;
        } else {
            this.checkpointPhase = CheckpointPhase.getPhase(System.getProperty("io.openliberty.ejb.checkpoint.phase", ""));
        }

        // For any Checkpoint phase, pause all non-persistent timers until checkpoint restored
        if (checkpointPhase != CheckpointPhase.INACTIVE && !checkpointPhase.restored()) {
            TimerNpRunnable.pause();
        }
    }
```
I can change it back to that.

Reply:
The old behavior would not call `pause()` if checkpoint phase was obtained from the property, rather than the real checkpoint service.  Because `checkpoint` is no longer passed in as a parameter, `checkpoint` is the same as `this.checkpoint`, whereas in the original code it was not.  In the original code, `checkpoint` could have been `null`, whereas `this.checkpoint` may have been APPLICATION (set from the property).

The above code is still not the same behavior as before.... instead, it would need to be something like

```
        boolean checkpointFromProperty = false;
        CheckpointPhase checkpointPhase = CheckpointPhase.getPhase();
        if (checkpointPhase != CheckpointPhase.INACTIVE || !ProductInfo.getBetaEdition()) {
            this.checkpointPhase = checkpointPhase;
        } else {
            checkpointFromProperty = true;
            this.checkpointPhase = CheckpointPhase.getPhase(System.getProperty("io.openliberty.ejb.checkpoint.phase", ""));
        }

        // For any Checkpoint phase, pause all non-persistent timers until checkpoint restored
        if (!checkpointFromProperty && !checkpointPhase.restored()) {
            TimerNpRunnable.pause();
        }
    }
```

Reply:
Ah, now I get it.  But with the change to how `INACTIVE` is always registered then we always do call `checkpointRestored` and this constructor would have set `this.checkpointPhase to the phase specified by the property which the tests would have set it to FEATURES, DEPLOYMENT or APPLICATIONS so the check for INACTIVE would pass making it call the `TimerNpRunnable.resume()`

This is a bit obscure.  I'll try to make this more straightforward.